src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs

```diff
@@ -306,23 +307,23 @@ public bool TryGet(ref Position position, out ReadOnlyMemory<byte> item, bool ad
                 if (position == Position.First)
                 {
                     item = _data;
-                    if (advance) { position.IntegerPosition++; position.ObjectPosition = _next; }
-                    return true;
+                    if (advance) {
+                        position.IntegerPosition++;
+                        position.ObjectPosition = _next;
+                    }
+                    return (!_data.IsEmpty || _next != null);
+                }
+                else if (position.ObjectPosition == null)
```

Comment:
Can we use `Cursor` instead of position? Seems to be the same concept.

Reply:
Yeah, I was thinking about it, but there are two things to consider:
1. Curstors point to Ts, Position points to segments. If we combine them, people will have to keep track which Position/Cursor is which.
2. Position has (and must have) three fields, Cursors have two. If the combined type ends up with three fields, it will be a bit slower to pass around and will make types that contain it bigger.

None the less, very much something we should consider.